python/plugins/processing/algs/grass7/Grass7Algorithm.py

```diff
@@ -839,28 +862,34 @@ def exportVectorLayerFromParameter(self, name, parameters, context):
                 if layerType in self.QGIS_OUTPUT_TYPES:
                     dataType = self.QGIS_OUTPUT_TYPES[layerType]
 
-        grassName = '{}{}'.format(name, self.uniqueSuffix)
-        self.exportVectorLayer(grassName, fileName, dataType)
+        outFormat = QgsVectorFileWriter.driverForExtension(os.path.splitext(fileName)[1])
+        dsco = self.parameterAsString(parameters, self.GRASS_VECTOR_DSCO, context)
+        lco = self.parameterAsString(parameters, self.GRASS_VECTOR_LCO, context)
+        self.exportVectorLayer(grassName, fileName, layer, nocats, dataType, outFormat, dsco, lco)
 
-    def exportVectorLayer(self, grassName, fileName, dataType='auto', layer=None, nocats=False):
+    def exportVectorLayer(self, grassName, fileName, layer=None, nocats=False, dataType='auto',
+                          outFormat='ESRI_Shapefile', dsco=None, lco=None):
```

Comment:
Is there a particular reason to still enforce ESRI_Shapefile?
IMHO the (meanwhile default) GPKG better keeps attribute names etc.

Reply:
Just looked at https://grass.osgeo.org/grass74/manuals/v.out.ogr.html and as I can see GPKG is not in the list of supported formats